### PR TITLE
[feat]: [PIE-8577]: added move config inputset in the openapi yaml

### PR DESCRIPTION
**Pull request by adivishy1**
## Describe your changes
Added move config inputset api in the openapi yaml and its impl. Have tested out the move config flow using tf apis and its working as expected.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44728)

### CODE REVIEW

**Review by mankrit-singh — DISMISSED** [review]
> Reviewed 3 of 7 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44728)* status: 9 file(s) to review. 1 of 2 approvals obtained.

**Review by vaibhav-si — APPROVED** [review]

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 3 of 7 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44728)* status: :shipit: complete! ✓

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -489,6 +489,27 @@ paths:
...
       x-internal: false
+  '/v1/orgs/{org}/projects/{project}/input-sets/{input-set}/move-config':
+    parameters:
+      - $ref: '#/components/parameters/OrgParam'
+      - $ref: '#/components/parameters/ProjectParam'
+      - $ref: '#/components/parameters/InputSetParam'
+    post:
+      summary: Move InputSet YAML from inline to remote or remote to inline
+      operationId: inputSetMoveConfig
+      description: Fetches Input-Set YAML from Harness DB and creates a remote entity or Fetches Input-Set YAML from remote repository and creates an inline entity
```
Comment (rashminandasahoo) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:500]:
```suggestion
      description: Creates a remote entity by fetching the input set YAML from Harness or an inline entity by fetching input set YAML from a remote repository.
```

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -1909,6 +1930,30 @@ components:
         repo_name:
           type: string
           description: Name of the repository.
+    InputSetMoveConfigRequestBody:
+      title: InputSetMoveConfigRequestBody
+      x-stoplight:
+        id: gf4xlhs210icv
+      type: object
+      description: 'Request body for input-set move configuration '
```
Comment (rashminandasahoo) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:1938]:
```suggestion
      description: 'Request body for moving an input set configuration.'
```

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 3]
```diff
@@ -1909,6 +1930,30 @@ components:
...
+          $ref: '#/components/schemas/MoveConfigOperationType'
+    InputSetMoveConfigResponseBody:
+      title: InputSetMoveConfigResponseBody
+      x-stoplight:
+        id: yk5yx84ss2dtu
+      type: object
+      properties:
+        input_set_identifier:
+          type: string
+      description: 'Response body for input-set move configuration '
```
Comment (rashminandasahoo) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:1956]:
```suggestion
      description: 'Response body for moving an input set configuration.'
```

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 4]
```diff
@@ -2495,6 +2540,12 @@ components:
         application/json:
           schema:
             $ref: '#/components/schemas/PipelineMoveConfigResponseBody'
+    InputSetMoveConfigResponseBody:
+      description: Response body for Input-Set move-config.
```
Comment (rashminandasahoo) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:2544]: Please refer to previous comments.

**Review by rashminandasahoo — CHANGES_REQUESTED** [review]

**Review by adivishy1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44728)* status: 5 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 500 at r3](https://reviewable.io/reviews/harness/harness-core/44728#-NPpaWxJbPAW8piBO_Nx-r3-500:-NPzXSTnDGhPFbQaCOxR:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a17b2957bcd9249931243db4aac0e439569358dc/pipeline-service/contracts/openapi/v1/openapi.yaml#L500)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
      description: Creates a remote entity by fetching the input set YAML from Harness or an inline entity by fetching input set YAML from a remote repository.
```
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 1938 at r3](https://reviewable.io/reviews/harness/harness-core/44728#-NPpaWxJbPAW8piBO_Nx-r3-1938:-NPzXa6JEU9upt2VUDAm:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a17b2957bcd9249931243db4aac0e439569358dc/pipeline-service/contracts/openapi/v1/openapi.yaml#L1938)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
      description: 'Request body for moving an input set configuration.'
```
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 1956 at r3](https://reviewable.io/reviews/harness/harness-core/44728#-NPpaWxJbPAW8piBO_Nx-r3-1956:-NPzXdUU8eSvdZ_-WSVN:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a17b2957bcd9249931243db4aac0e439569358dc/pipeline-service/contracts/openapi/v1/openapi.yaml#L1956)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
      description: 'Response body for moving an input set configuration.'
```
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 2544 at r3](https://reviewable.io/reviews/harness/harness-core/44728#-NPpaWxJbPAW8piBO_Nx-r3-2544:-NPzXm7k98Lemcl8Ksns:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a17b2957bcd9249931243db4aac0e439569358dc/pipeline-service/contracts/openapi/v1/openapi.yaml#L2544)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Please refer to previous comments.
</blockquote></details>

Done.

**Review by rashminandasahoo — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44728)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by mankrit-singh — APPROVED** [review]
> Reviewed 4 of 5 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44728)* status: 5 file(s) to review.

**Review by mankrit-singh — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44728#-:-NPz_L989LL8iMtrnKoN:bnfp4nl)* status: 5 file(s) to review.

**Comment by adivishy1** [comment]
trigger ut3

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 4 of 6 files at r2, 4 of 5 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44728)* status: :shipit: complete! ✓

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 4 of 6 files at r2, 4 of 5 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44728)* status: :shipit: complete! ✓

**Pull request merged**